Assignment3A/StringManipulation.java

```diff
@@ -0,0 +1,185 @@
+package Assignment3A;
+
+/**
+ * 
+ * this class contain methods to manipulate strings we can perform
+ * comparison,reverse of string, upper case to lower case etc
+ *
+ */
+public class StringManipulation {
+
+	/**
+	 * To compare two strings
+	 * @param firstString contain a not null case sensitive String
+	 * @param secondString contain a not null case sensitive String
+	 * @return Integer value 1 if strings are equal and 0 if not
+	 */
+	public int compareStrings(String firstString, String secondString) {
+
+		char firstStringCharArray[] = firstString.toCharArray();
+		int firstStringLength = firstString.length();
+
+		char secondStringCharArray[] = secondString.toCharArray();
+		int secondStringLength = secondString.length();
+
+		if (firstStringLength != secondStringLength) {
+			return 0;
+		} else {
+			int charIndex = 0;
+			while (charIndex != firstStringLength - 1) {
+				if (firstStringCharArray[charIndex] != secondStringCharArray[charIndex]) {
+					return 0;
+				}
+				charIndex++;
+			}
+		}
+		return 1;
+	}
+
+	/**
+	 * To get reverse of a string
+	 * @param firstString contain a not null case sensitive String
+	 * @return reverse of string
+	 */
+	public String reverseOfString(String firstString) {
+		char firstStringCharArray[] = firstString.toCharArray();
+		int firstStringLength = firstString.length();
+
+		char reverseOfString[] = new char[firstStringLength];
+
+		int reverseCharIndex = firstStringLength - 1;
+		int frontCharIndex = 0;
+		while (reverseCharIndex >= 0) {
+			reverseOfString[frontCharIndex] = firstStringCharArray[reverseCharIndex];
+			reverseCharIndex--;
+			frontCharIndex++;
+		}
+		String reverseString = new String(reverseOfString);
+		return reverseString;
+	}
+
+	/**
+	 * To convert upper case character to lower case and vice-versa
+	 * @param firstString contain a not null case sensitive String
+	 * @return String thats character converted from lower case to upper case and vice-versa
+	 */
+	public String chracterConversionOfString(String firstString) {
+		char firstStringCharArray[] = firstString.toCharArray();
+		int firstStringLength = firstString.length();
+
+		char convertedString[] = new char[firstStringLength];
+
+		int charIndex = 0;
+
+		while (charIndex < firstStringLength) {
+
+			if (firstStringCharArray[charIndex] >= 'A' && firstStringCharArray[charIndex] <= 'Z') {
+				convertedString[charIndex] = (char) (firstStringCharArray[charIndex] + 32);

```

Comment:
Please create constant for specific value. Here like character buffer 32 should be a constant variable.